wire-runtime/src/commonMain/kotlin/com/squareup/wire/ProtoReader.kt

```diff
@@ -391,29 +412,22 @@ class ProtoReader(private val source: BufferedSource) {
   /** Reads each tag, handles it, and returns a byte string with the unknown fields. */
   @JvmName("-forEachTag") // hide from Java
   inline fun forEachTag(tagHandler: (Int) -> Any): ByteString {
-    // Lazily created if the current message has unknown fields.
-    var unknownFieldsBuffer: Buffer? = null
-    var unknownFieldsWriter: ProtoWriter? = null
-
     val token = beginMessage()
-    var tag: Int
     while (true) {
-      tag = nextTag()
+      val tag = nextTag()
       if (tag == -1) break
-      if (tagHandler(tag) !== UNKNOWN_TAG) continue
-      if (unknownFieldsBuffer == null) {
-        unknownFieldsBuffer = Buffer()
-        unknownFieldsWriter = ProtoWriter(unknownFieldsBuffer)
-      }
-      val fieldEncoding = peekFieldEncoding()
-      val protoAdapter = fieldEncoding!!.rawProtoAdapter()
-      val value = protoAdapter.decode(this)
-      @Suppress("UNCHECKED_CAST") // We encode and decode the same types.
-      (protoAdapter as ProtoAdapter<Any>).encodeWithTag(unknownFieldsWriter!!, tag, value)
+      tagHandler(tag)
     }
-    endMessage(token)
+    return endMessageAndGetUnknownFields(token)
+  }
 
-    return unknownFieldsBuffer?.readByteString() ?: ByteString.EMPTY
+  fun readUnknownField(tag: Int) {
```

Comment:
Document that the unknown field will be included in endMessageAndGetUnknownFields’s return value?